lib/widgets/profile.dart

```diff
@@ -0,0 +1,334 @@
+        final Map<String, dynamic> fieldData = jsonDecode(entry.fieldData);
+        String? urlPattern = fieldData["url_pattern"];
+        if (urlPattern == null) {
+          final RealmDefaultExternalAccount? realmDefaultExternalAccount = store.realmDefaultExternalAccounts[fieldData["subtype"]];
+          if (realmDefaultExternalAccount != null) {
+            urlPattern = realmDefaultExternalAccount.urlPattern;
+          }
+        }
+        if (urlPattern != null) {
+          final url = urlPattern.replaceFirst("%(username)s", entry.text);
+          return Padding(
+            padding: const EdgeInsets.all(4),
+            child: ProfileDataTableLink(url: url, label: entry.text));
+        } else {
+          return Padding(
+            padding: const EdgeInsets.all(4),
+            child: Text(entry.text));
+        }
+      case CustomProfileFieldType.user:
+        // TODO(server): This is completely undocumented.  The key to
+        //   reverse-engineering it was:
+        //   https://github.com/zulip/zulip/blob/18230fcd9/static/js/settings_account.js#L247
+        final List<dynamic> userIds = jsonDecode(entry.text);
+        if (userIds.isEmpty) {
+          return const SizedBox.shrink();
+        }
+        return Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: userIds.whereType<num>()
+            .map((userId) => store.users[userId])
+            .whereType<User>()
+            .map((user) => ProfileDataTableUser(user: user)).toList());
+      case CustomProfileFieldType.date:
+      // TODO(server): The value's format is undocumented, but empirically
+      //   it's a date in ISO format, like 2000-01-01.
+      // That's readable as is, but:
+      // TODO format this date using user's locale.
+        return Padding(
+          padding: const EdgeInsets.all(4),
+          child: Text(entry.text));
+      case CustomProfileFieldType.shortText:
+      case CustomProfileFieldType.longText:
+      case CustomProfileFieldType.pronouns:
+      case CustomProfileFieldType.unknown:
+        return Padding(
+          padding: const EdgeInsets.all(4),
+          child: Text(entry.text));
+    }
+  }
+}
+
+class CustomProfileFieldDisplayEntry {
+  final String label;
+  final String text;
+  final CustomProfileFieldType type;
+  final String fieldData;
+
+  CustomProfileFieldDisplayEntry({required this.fieldData, required this.label, required this.text, required this.type});
+}
+
+class ProfileDataTableLink extends StatelessWidget {
+  const ProfileDataTableLink({super.key, required this.label, required this.url});
+
+  final String label;
+  final String url;
+
+  @override
+  Widget build(BuildContext context) {
+    final Uri? parsedUrl = Uri.tryParse(url);
+    if (parsedUrl == null) {
+      return Text(label);
+    }
+    return MouseRegion(
+      cursor: SystemMouseCursors.click,
+      child: GestureDetector(
```

Comment:
Hmm interesting. Perhaps we should be doing this on other links too? In particular in message content.

Reply:
I think it's not likely that users will have a mouse cursor on mobile (although I think that's possible with bluetooth devices nowadays) but I think it's a nice touch to add!